src/facade/dragonfly_connection.cc

```diff
@@ -300,22 +309,23 @@ void Connection::HandleRequests() {
 
   auto remote_ep = lsb->RemoteEndpoint();
 
+  FiberSocketBase* peer = socket_.get();
 #ifdef DFLY_USE_SSL
   unique_ptr<tls::TlsSocket> tls_sock;
   if (ctx_) {
-    tls_sock.reset(new tls::TlsSocket(socket_.get()));
-    tls_sock->InitSSL(ctx_);
-
-    FiberSocketBase::AcceptResult aresult = tls_sock->Accept();
-    if (!aresult) {
-      LOG(WARNING) << "Error handshaking " << aresult.error().message();
-      return;
+    const bool no_tls_on_admin_port = absl::GetFlag(FLAGS_no_tls_on_admin_port);
+    if (!no_tls_on_admin_port || (no_tls_on_admin_port && !IsAdmin())) {
+      tls_sock.reset(new tls::TlsSocket(socket_.get()));
+      tls_sock->InitSSL(ctx_);
+      FiberSocketBase::AcceptResult aresult = tls_sock->Accept();
+
+      if (!aresult) {
+        LOG(WARNING) << "Error handshaking " << aresult.error().message();
+        return;
+      }
+      VLOG(1) << "TLS handshake succeeded";
     }
-    VLOG(1) << "TLS handshake succeeded";
   }
-  FiberSocketBase* peer = tls_sock ? (FiberSocketBase*)tls_sock.get() : socket_.get();
```

Comment:
why was this removed?

Reply:
Because I simplified it a little bit, it gets initialized at the top :)

Reply:
how is it working?
I would expect the test you wrote to fail.

Reply:
Indeed I messed this up on refactoring and HUGE thumps up for catching this. It works, because we only test the admin port and not the TLS connection itself -- we have 0 tests for that (although my other PR addresses this and it would have been caught there).